File: MovieCatalogue_Java_Submission_1/app/src/main/java/id/apwdevs/moviecatalogue/adapter/TopBilledCastAdapter.java
package id.apwdevs.moviecatalogue.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import id.apwdevs.moviecatalogue.R;
import id.apwdevs.moviecatalogue.model.FilmTopBilledCastModel;

public class TopBilledCastAdapter extends BaseAdapter {

    private List<FilmTopBilledCastModel> filmTopBilledCastModels;
    private Context mContext;

    public TopBilledCastAdapter(Context mContext) {
        this.mContext = mContext;
        filmTopBilledCastModels = new ArrayList<>();
    }

    public void setFilmTopBilledCastModels(List<FilmTopBilledCastModel> filmTopBilledCastModels) {
        this.filmTopBilledCastModels = filmTopBilledCastModels;
    }

    @Override
    public int getCount() {
        return filmTopBilledCastModels.size();
    }

    @Override
    public Object getItem(int position) {
        return filmTopBilledCastModels.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null)
            convertView = LayoutInflater.from(mContext).inflate(R.layout.item_list_actor, parent, false);

        FilmTopBilledCastModel filmTopBilledCastModel = (FilmTopBilledCastModel) getItem(position);

        TopBilledCastViewHolder topBilledCastViewHolder = new TopBilledCastViewHolder(convertView);
        topBilledCastViewHolder.bind(filmTopBilledCastModel);
        return convertView;
    }

    private class TopBilledCastViewHolder {
        private TextView name, alias;

        TopBilledCastViewHolder(View view) {
            name = view.findViewById(R.id.item_list_actor_name);
            alias = view.findViewById(R.id.item_list_actor_alias);
        }

        void bind(FilmTopBilledCastModel filmTopBilledCastModel) {
            name.setText(filmTopBilledCastModel.getActorName());
            alias.setText(filmTopBilledCastModel.getAliasPeople());
        }
    }
}

File: MovieCatalogue_Kotlin_Submission_3/app/src/main/java/id/apwdevs/moTvCatalogue/adapter/TopBilledCastAdapter.kt
package id.apwdevs.moTvCatalogue.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import id.apwdevs.moTvCatalogue.R
import id.apwdevs.moTvCatalogue.model.FilmTopBilledCastModel

class TopBilledCastAdapter(private val mContext: Context) :
    RecyclerView.Adapter<TopBilledCastAdapter.TopBilledCastViewHolder>() {
    var filmTopBilledCastModels: List<FilmTopBilledCastModel>? = null


    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): TopBilledCastViewHolder {
        val view = LayoutInflater.from(mContext).inflate(R.layout.item_list_actor, p0, false)
        return TopBilledCastViewHolder(view)
    }

    override fun getItemCount(): Int {
        return when (val itemCount = filmTopBilledCastModels?.size) {
            null -> 0
            else -> itemCount
        }
    }

    override fun onBindViewHolder(p0: TopBilledCastViewHolder, p1: Int) {
        val filmCastModel = filmTopBilledCastModels?.get(p1) ?: return
        p0.bind(filmCastModel)
    }

    inner class TopBilledCastViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(view) {
        private val name: TextView = view.findViewById(R.id.item_list_actor_name)
        private val alias: TextView = view.findViewById(R.id.item_list_actor_alias)

        internal fun bind(filmTopBilledCastModel: FilmTopBilledCastModel) {
            name.text = filmTopBilledCastModel.actorName
            alias.text = filmTopBilledCastModel.aliasPeople
        }
    }
}